src/algebra/big_operators.lean

```diff
@@ -462,6 +462,12 @@ lemma prod_range_succ' (f : ℕ → β) :
 | (n + 1) := by rw [prod_range_succ (λ m, f (nat.succ m)), mul_assoc, ← prod_range_succ'];
                  exact prod_range_succ _ _
 
+lemma le_sum_nat_of_mem (f : ℕ → ℕ) (s : finset ℕ) (a : ℕ) (t : a ∈ s) : f a ≤ ∑ i in s, f i :=
```

Comment:
You could generalize the source of `f` to any `Type*`.

Reply:
I think you can generalize the target of `f` to any `[canonically_ordered_add_monoid M]`?

Reply:
Only if we can somehow show that everything in the image of `s` under `f` is nonnegative. It's a bit of a nasty change, pollutes the type signature a lot.

Reply:
I'll just use `single_le_sum` instead.